### PR TITLE
Snippets and tests for Firestore listens

### PR DESCRIPTION
**Pull request by samtstern**
Note: this can't be merged until `0.32.1` is release, I wrote the PR against a `SNAPSHOT` build.

### CODE REVIEW

**Comment by samtstern** [comment]
cc @schmidt-sebastian just FYI

File: firestore/src/test/java/com/example/firestore/BaseIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.cloud.firestore.QuerySnapshot;
+import java.util.Map;
+import org.junit.BeforeClass;
+
+/**
+ * Base class for tests like {@link ManageDataSnippetsIT}.
+ */
+public class BaseIntegrationTest {
+
+  protected static String projectId = "java-docs-samples-firestore";
```
[the remark this answers is not in the document]
Comment (samtstern) [reply firestore/src/test/java/com/example/firestore/BaseIntegrationTest.java]: @lesv sorry not sure what you mean by "make sure this isn't listed in the parent pom"?  Also this project ID was added by @jabubake I just moved it into a new base class.

Side note: the tests failed but that's expected due to the `SNAPSHOT` build of the SDK.  Waiting for the next `google-cloud-java` release.

**Comment by samtstern** [comment]
@lesv the SDK has been released so we can test this now.

**Comment by samtstern** [comment]
Looks like the failure was `checkstyle` related.  I have one remaining style issue I don't know about:

```
/usr/local/google/home/samstern/Projects/snippets/cloud/java/firestore/src/test/java/com/example/firestore/snippets/ListenDataSnippetsIT.java:36: Abbreviation in name 'ListenDataSnippetsIT' must contain no more than '2' consecutive capital letters. [AbbreviationAsWordInName]
```

How does the name `ListenDataSnippetsIT` violate this?

**Comment by lesv** [comment]
Looks like you just need to suppress Warnings as you did in the other IT's.  It's from the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html) btw - this looks like section 5.3.

**Comment by lesv** [comment]
@kurtisvg Next time your messing w/ java-repo-tools (the style guide, see if you can special case the IT suffix.

**Comment by kurtisvg** [comment]
The short term fix is that we can do the following:
```java
//CHECKSTYLE OFF: AbbreviationAsWordInName
//CHECKSTYLE ON: AbbreviationAsWordInName
```
As long as the violating line in question is between those lines (ie, class declaration), it won't error out. (this is the process for ignore specific checkstyle errors in general) 

I can specifically disable the "AbbreviationAsWordInName" requirement, but I'll need to look into it to see if I can allow 'IT' specifically.

**Comment by samtstern** [comment]
@lesv ok I think this one is finally ready!

**Comment by samtstern** [comment]
Looks like there was an unrelated failure?

```
[INFO] Google Cloud Firestore Samples ..................... SUCCESS [ 17.861 s]
```

```
[INFO] appengine-guestbook-cloud-datastore-j8 ............. FAILURE [ 15.463 s]
```

**Pull request merged**